src/Futhark/CLI/LSP.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExplicitNamespaces #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Futhark.CLI.LSP (main) where
+
+import Control.Concurrent.MVar (newMVar)
+import Control.Monad.IO.Class (MonadIO (liftIO))
+import Futhark.LSP.Handlers (handlers)
+import Futhark.LSP.State (emptyState)
+import Futhark.Util (debug)
+import Language.LSP.Server
```

Comment:
OK, seeing how many imports this needs, it's getting a bit silly.  Half of the file becomes imports!  It's probably better to move it back to how it used to be.